be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -98,29 +98,34 @@ Status BetaRowsetWriter::init(const RowsetWriterContext& rowset_writer_context)
 }
 
 Status BetaRowsetWriter::add_block(const vectorized::Block* block) {
-    if (block->rows() == 0) {
-        return Status::OK();
-    }
     if (UNLIKELY(_segment_writer == nullptr)) {
         RETURN_NOT_OK(_create_segment_writer(&_segment_writer));
     }
+    return _add_block(block, &_segment_writer);
+}
+
+Status BetaRowsetWriter::_add_block(const vectorized::Block* block,
+                                    std::unique_ptr<segment_v2::SegmentWriter>* segment_writer) {
```

Comment:
do not pass unique_ptr<T>* any more, just pass in unique_ptr<T>&

Reply:
add block will change segment_writer so i use unique_ptr*